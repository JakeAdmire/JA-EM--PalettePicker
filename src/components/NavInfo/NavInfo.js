import React, { Component } from 'react';

export class NavInfo extends Component {

  mapProjectInfo = () => {
    let mappedProjects = this.props.projects.map(project => (
      <div  className="project-container" 
            key={project.id} >

        <h6>TITLE:</h6>
        <p key={project.id}>{project.title}</p>
        <button   className="project-delete" 
                  value="projects" 
                  id={project.id} 
                  onClick={this.props.handleDelete}>X</button>

        <div>
          <h6>PALETTES:</h6>
          {
            project.palettes.length
              ? this.mapPaletteInfo(project)
              : <p>Add a palette!</p>
          } 
        </div>
      </div>
      )
    )

    return mappedProjects;
  }

  mapPaletteInfo = (project) => {
    let mappedPalettes = project.palettes.map(palette => (
      <div  key={palette.id} 
            className="palette-container" >
      
        <p id="paletteEdit" onClick={() => this.props.setHex([
          {hex: palette.color1}, 
          {hex: palette.color2}, 
          {hex: palette.color3}, 
          {hex: palette.color4}, 
          {hex: palette.color5}
          ])}>{palette.name}</p>

        <div className="palette-buttons">

          <button   className="palette-save" 
                    value="palettes" 
                    id={palette.id} 
                    onClick={this.props.handlePatch}>SAVE</button>

          <button   className="palette-delete" 
                    value="palettes" 
                    id={palette.id} 
                    onClick={this.props.handleDelete}>X</button>
        </div>
      </div>
      )
    )
    return mappedPalettes;
  }

  render() {

    return (

      <div className="info-section projects">

        {
          this.props.projects.length
            ? this.mapProjectInfo()
            : <p className="splash-text">â—„ Try Creating a Project!</p>
        }
        
      </div>

    )
  }

}
