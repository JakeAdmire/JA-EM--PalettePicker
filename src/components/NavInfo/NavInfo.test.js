import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';

import { NavInfo } from './NavInfo';

describe('NavInfo', () => {

  let projectsMock = [
    { 
      title: "Test Project", 
      id: 73, 
      palettes: [
        {
          color1: "#7DBB54",
          color2: "#569963",
          color3: "#486A60",
          color4: "#455154",
          color5: "#313235",
          id: 63,
          name: "Test Palette",
          project_id: 73
        }
      ] 
    }
  ];
  let handleDeleteMock = jest.fn();
  let handlePatchMock = jest.fn();
  let setHexMock = jest.fn();

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <NavInfo  projects={projectsMock}
                handleDelete={handleDeleteMock}
                handlePatch={handlePatchMock}
                setHex={setHexMock} />
    )
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it.skip('should invoke mapProjectInfo on mount', () => {
    const mockSpy = jest.spyOn(wrapper.instance(), 'mapProjectInfo');

    wrapper.update();

    expect(mockSpy).toHaveBeenCalled();
  })

  describe('mapProjectInfo()', () => {

    it('should return JSX', () => {
      let results = wrapper.instance().mapProjectInfo();

      expect(results).toMatchSnapshot();
    })

    it('should invoke mapPaletteInfo', () => {
      const mockSpy = jest.spyOn(wrapper.instance(), 'mapPaletteInfo');

      wrapper.instance().mapProjectInfo();

      expect(mockSpy).toHaveBeenCalled();
    })

  })

  describe('mapPaletteInfo()', () => {

    it('should return JSX', () => {
      let results = wrapper.instance().mapPaletteInfo(projectsMock[0]);

      expect(results).toMatchSnapshot();
    })

  })

})