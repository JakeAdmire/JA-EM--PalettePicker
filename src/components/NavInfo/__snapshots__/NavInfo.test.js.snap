// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavInfo mapPaletteInfo() should return JSX 1`] = `
Array [
  <div
    className="palette-container"
  >
    <p
      id="paletteEdit"
      onClick={[Function]}
    >
      Test Palette
    </p>
    <div
      className="palette-buttons"
    >
      <button
        className="palette-save"
        id={63}
        onClick={[MockFunction]}
        value="palettes"
      >
        SAVE
      </button>
      <button
        className="palette-delete"
        id={63}
        onClick={[MockFunction]}
        value="palettes"
      >
        X
      </button>
    </div>
  </div>,
]
`;

exports[`NavInfo mapProjectInfo() should return JSX 1`] = `
Array [
  <div
    className="project-container"
  >
    <h6>
      TITLE:
    </h6>
    <p>
      Test Project
    </p>
    <button
      className="project-delete"
      id={73}
      onClick={[MockFunction]}
      value="projects"
    >
      X
    </button>
    <div>
      <h6>
        PALETTES:
      </h6>
      <div
        className="palette-container"
      >
        <p
          id="paletteEdit"
          onClick={[Function]}
        >
          Test Palette
        </p>
        <div
          className="palette-buttons"
        >
          <button
            className="palette-save"
            id={63}
            onClick={[MockFunction]}
            value="palettes"
          >
            SAVE
          </button>
          <button
            className="palette-delete"
            id={63}
            onClick={[MockFunction]}
            value="palettes"
          >
            X
          </button>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`NavInfo should match the snapshot 1`] = `ShallowWrapper {}`;
